File: junit5-vanilla-modules/src/test/com.example.tool/com/example/tool/CalculatorTests.java
/*
 * Copyright 2015-2018 the original author or authors.
 *
 * All rights reserved. This program and the accompanying materials are
 * made available under the terms of the Eclipse Public License v2.0 which
 * accompanies this distribution and is available at
 *
 * http://www.eclipse.org/legal/epl-v20.html
 */

package com.example.tool;

import org.junit.jupiter.api.*;

@DisplayName("com.example.tool/com.example.tool.CalculatorTests")
class CalculatorTests {

	private final Calculator calculator = new Calculator();

	@Test
	void add() {
		Assertions.assertEquals(2, calculator.add(1, 1), "1 + 1 should equal 2");
	}

	@Test
	void mul() {
		Assertions.assertEquals(4, calculator.mul(2, 2), "2 * 2 should equal 4");
	}

	@Test
	void pow() {
		Assertions.assertEquals(8, calculator.pow(2, 3), "2 ^ 3 should equal 8");
	}

}

File: junit5-jupiter-starter-maven-scala/src/test/scala/com/example/project/CalculatorTests.scala
/*
 * Copyright 2015-2018 the original author or authors.
 *
 * All rights reserved. This program and the accompanying materials are
 * made available under the terms of the Eclipse Public License v2.0 which
 * accompanies this distribution and is available at
 *
 * http://www.eclipse.org/legal/epl-v20.html
 */

package com.example.project

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.{DisplayName, Test}

class CalculatorTests {

	@Test
	@DisplayName("should add two numbers")
	def test(): Unit = Seq(
		(0, 1, 1),
		(1, 1, 2),
		(1, 2, 3),
		(49, 51, 100),
		(1, 100, 101),
	) collect {
		case Tuple3(a, b, result) =>
			assertEquals(result, Calculator.add(a, b), s"$a + $b should equal $result")
	}
}
